awx/ui_next/src/screens/Credential/CredentialEdit/CredentialEdit.jsx

```diff
@@ -12,45 +16,100 @@ function CredentialEdit({ credential, me }) {
   const [error, setError] = useState(null);
   const [isLoading, setIsLoading] = useState(true);
   const [credentialTypes, setCredentialTypes] = useState(null);
+  const [inputSources, setInputSources] = useState(null);
   const [formSubmitError, setFormSubmitError] = useState(null);
   const history = useHistory();
 
   useEffect(() => {
     const loadData = async () => {
       try {
-        const {
-          data: { results: loadedCredentialTypes },
-        } = await CredentialTypesAPI.read({
-          or__namespace: ['gce', 'scm', 'ssh'],
-        });
+        const [
+          {
+            data: { results: loadedCredentialTypes },
+          },
+          {
+            data: { results: loadedInputSources },
+          },
+        ] = await Promise.all([
+          CredentialTypesAPI.read({
+            or__namespace: ['gce', 'scm', 'ssh'],
+          }),
+          CredentialsAPI.readInputSources(credential.id, { page_size: 200 }),
+        ]);
         setCredentialTypes(loadedCredentialTypes);
+        const inputSourcesMap = {};
```

Comment:
I think you could use an actual map here
const inputSourcesMap = loadedInputSources.map(inputSource => ({inputSource.input_field_name: inputSource}))

Reply:
^^ @jlmitch5 and I talked about this one - going with a `.reduce` solution